multinet/auth/types.py

```diff
@@ -0,0 +1,73 @@
+"""Types associated with authentication."""
+
+from typing import Optional
+from dataclasses import dataclass
+
+
+@dataclass
+class GoogleUserInfo:
+    """
+    Representation of raw User Info from a Google Account.
+
+    Mostly unused by us, but is the data type of the decoded ID Token.
+    Reference: https://developers.google.com/identity/protocols/oauth2/openid-connect#an-id-tokens-payload  # noqa
+    """
+
+    family_name: str
+    given_name: str
+    name: str
+    picture: str
+    locale: str
+    sub: str  # Unique identifier
+    email: str
```

Comment:
We might not be able to do this. It will muddy up the types in a way that we wouldn't want. Basically, we could return or send in the wrong type from/to a function that expects the other, and the type system will allow it.

That's my speculation at least. It needs to be tested to see if that is actually as I think it is.